Sources/SwiftProtobuf/JSONEncodingVisitor.swift

```diff
@@ -274,12 +285,35 @@ internal struct JSONEncodingVisitor: Visitor {
   }
 
   mutating func visitRepeatedMessageField<M: Message>(value: [M], fieldNumber: Int) throws {
-    let localOptions = options
-    try _visitRepeated(value: value, fieldNumber: fieldNumber) {
-      (encoder: inout JSONEncoder, v: M) throws in
-      let json = try v.jsonUTF8Data(options: localOptions)
-      encoder.append(utf8Data: json)
+    try startField(for: fieldNumber)
+    var comma = false
+    encoder.startArray()
+    if let _ = M.self as? _CustomJSONCodable.Type {
+      for v in value {
+        if comma {
+          encoder.comma()
+        }
+        comma = true
+        let json = try v.jsonString(options: options)
+        encoder.append(text: json)
+      }
+    } else if let newNameMap = (M.self as? _ProtoNameProviding.Type)?._protobuf_nameMap {
+      let oldNameMap = self.nameMap
+      self.nameMap = newNameMap
+      for v in value {
+        if comma {
+          encoder.comma()
+        }
+        comma = true
+        encoder.startNestedObject()
+        try v.traverse(visitor: &self)
+        encoder.endObject()
+      }
+      self.nameMap = oldNameMap
```

Comment:
Could do this with a `defer` right after swapping it so it is less likely to get missed in any other changes.

Reply:
Good catch!  Done.